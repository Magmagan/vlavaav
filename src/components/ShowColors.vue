<template>
  <div class="hello">
    <div
      v-for="color in colors"
      :key="color"
      :style="`background-color: ${color}; flex-grow: 1`"
      class="color"
    >
      <p>
        {{ color }}
      </p>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';

@Component
export default class ShowColors extends Vue {
  @Prop({ required: false, type: Array, default: () => [] }) readonly colors!: string[]
}
</script>

<style scoped lang="scss">
.hello {
  height: 100%;
  display: flex;
  flex-direction: column;
  width: 100%;
}
</style>
