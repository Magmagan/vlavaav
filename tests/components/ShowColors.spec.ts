import { shallowMount } from '@vue/test-utils';
import ShowColors from '@/components/ShowColors.vue';

describe('InitialInfo', () => {
  def('colors', () => ['white', 'yellow']);

  const wrapper = () => shallowMount(
    ShowColors, {
      propsData: {
        colors: get('colors'),
      },
    },
  );

  it('matches snapshot', () => {
    expect(wrapper().element).toMatchSnapshot();
  });

  describe('numberOfColumns', () => {
    describe('when the number of colors is a perfect square', () => {
      def('colors', () => [
        'white', 'yellow',
        'blue', 'green',
      ]);

      it('retuns the exact amount of columns needed', () => {
        expect(wrapper().vm.numberOfColumns).toEqual(2);
      });
    });

    describe('when the number of colors is bigger than a perfect square', () => {
      def('colors', () => [
        'white', 'yellow',
        'blue', 'green',
        'orange',
      ]);

      it('retuns an extra column', () => {
        expect(wrapper().vm.numberOfColumns).toEqual(3);
      });
    });
  });
});
