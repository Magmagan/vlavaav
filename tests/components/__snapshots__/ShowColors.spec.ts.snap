// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialInfo matches snapshot 1`] = `
<div>
  <div
    class="show-colors"
    style="grid-template-columns: repeat(2, 1fr);"
  >
    <div
      class="show-colors__cell"
      style="background-color: white; flex-grow: 1;"
    >
      <div
        class="show-colors__cell-description-box"
      >
        <code
          class="show-colors__cell-description"
        >
          
          white
          
          <br />
        </code>
      </div>
    </div>
    <div
      class="show-colors__cell"
      style="background-color: yellow; flex-grow: 1;"
    >
      <div
        class="show-colors__cell-description-box"
      >
        <code
          class="show-colors__cell-description"
        >
          
          yellow
          
          <br />
        </code>
      </div>
    </div>
  </div>
   
  <div
    class="show-colors__footer"
  >
    <a
      class="show-colors__footer-link"
      href="."
    >
      Click here
    </a>
    
    to paste a new set of colors.
    
    <strong>
      VLAVAAV
    </strong>
     Â© Alex L. Alves 2020.
  
  </div>
</div>
`;
